### PR TITLE
Add IsValidRegistry, Ensure ports flow with registries

### PR DESCRIPTION
**Pull request by BenVillalobos**
Merge after https://github.com/rainersigwald/containers/pull/54

### CODE REVIEW

**Comment by BenVillalobos** [comment]
The issue here is that `ContainerRegistry` isn't being set in the test, which results in an NRE.

TODO:
1. Add a null check in the `IsValidRegistry` function
2. Set the value in the test.

I'll wait until we merge a few more PR's to rebase this one.

File: System.Containers.Tasks/ParseContainerProperties.cs
Comment (baronfel) [inline System.Containers.Tasks/ParseContainerProperties.cs:77]: If no Scheme is supplied on the given URI, we should infer `https://`. This matches the behavior of when you are pushing images/tags to a registry, as well as what users use when you say `FROM <baseImage>` in a Dockerfile.

Comment (BenVillalobos) [reply System.Containers.Tasks/ParseContainerProperties.cs:85]: fixed

**Comment by BenVillalobos** [comment]
Rebased onto main

File: Microsoft.NET.Build.Containers/ParseContainerProperties.cs
Comment (rainersigwald) [inline Microsoft.NET.Build.Containers/ParseContainerProperties.cs:81]: Nit: you can get a permalink to a specific commit by hitting `y` on any source page on GitHub, and those are better in case they move stuff around/refactor.

```suggestion
            // Default to https when no scheme is present: https://github.com/distribution/distribution/blob/26163d82560f4dda94bd7b87d587f94644c5af79/reference/normalize.go#L88
```

File: Microsoft.NET.Build.Containers/ParseContainerProperties.cs
Comment (rainersigwald) [inline Microsoft.NET.Build.Containers/ParseContainerProperties.cs:87]: maybe

```suggestion
            Log.LogError("Could not recognize registry '{0}'. Is your registry missing 'https://'?", ContainerRegistry);
```

?

File: Microsoft.NET.Build.Containers/ParseContainerProperties.cs
Comment (baronfel) [inline Microsoft.NET.Build.Containers/ParseContainerProperties.cs:82]: nit: overwriting the users' means that if we issue an error below we show them something other than what they've entered. Can you use an interior variable to represent the changed value instead?

Comment (baronfel) [reply Microsoft.NET.Build.Containers/ParseContainerProperties.cs:82]: Also nit: can a test case be added that covers the "user didn't add a scheme but we still handled it" pathway for this task? so like `mcr.microsoft.com/dotnet/runtime:6.0` would end up with parsed properties of `https://mcr.microsoft.com", `dotnet/runtime`, and `6.0`.

Comment (BenVillalobos) [reply Microsoft.NET.Build.Containers/ParseContainerProperties.cs]: Donezo

**Review by rainersigwald — APPROVED** [review]

**Review by baronfel — APPROVED** [review]

**Pull request merged**
